From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: wangxyper <wangxyper@163.com>
Date: Sun, 1 Oct 2023 16:11:11 +0800
Subject: [PATCH] Fix rebrand in Versioning


diff --git a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
index 8a42351f06f0058431c1426527736d01130cc05a..e302f332fc40b9b6d78ffffe69ff77443a59120d 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
@@ -11,7 +11,7 @@ public final class Versioning {
     public static String getBukkitVersion() {
         String result = "Unknown-Version";
 
-        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/me.earthme.sandFolia/sandfolia-api/pom.properties"); // Folia //SandFolia
+        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/org.era4funmc.sandfolia/sandfolia-api/pom.properties"); // Folia //SandFolia
         Properties properties = new Properties();
 
         if (stream != null) {
